tutorials/dapt-curation/code/main.py

```diff
@@ -0,0 +1,245 @@
+    download_github_sources,
+    download_pdf_sources,
+    download_wikipedia_sources,
+)
+from utils import (
+    clean_and_unify,
+    dedupe,
+    filter_code,
+    filter_code_dataset,
+    filter_code_lines,
+    filter_txt,
+    filter_txt_lines,
+    redact,
+    redact_pii,
+)
+
+import nemo_curator as nc
+from nemo_curator import ScoreFilter, Sequential
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.filters import RepeatingTopNGramsFilter, WordCountFilter
+from nemo_curator.modifiers.pii_modifier import PiiModifier
+from nemo_curator.modifiers.unicode_reformatter import UnicodeReformatter
+from nemo_curator.modules import ExactDuplicates
+from nemo_curator.modules.modify import Modify
+from nemo_curator.utils.distributed_utils import get_client
+from nemo_curator.utils.file_utils import (
+    get_all_files_paths_under,
+    separate_by_metadata,
+)
+from nemo_curator.utils.script_utils import add_distributed_args, parse_client_args
+
+SCRIPT_DIR_PATH = os.path.dirname(os.path.abspath(__file__))
+DATA_DIR = os.path.join(SCRIPT_DIR_PATH, "data")
+
+#
+# TODO
+# Add instructions for apt install tessaract-ocr
+#
+
+
+def download_sources(
+    wikipedia_limit: Optional[int] = None,
+    github_limit: Optional[int] = None,
+    pdf_limit: Optional[int] = None,
+) -> Tuple[str, str]:
+    """
+    Downloads all the dataset sources and converts them to the JSONL format.
+
+    Returns:
+        tuple: the list of text files and the list of code files.
+    """
+
+    wikipedia_dir = download_wikipedia_sources(
+        "sources/wikipedia_urls.jsonl", limit=wikipedia_limit
+    )
+    github_dir = download_github_sources(
+        "sources/github_repos.jsonl", limit=github_limit
+    )
+    pdf_dir = download_pdf_sources("sources/arxiv_urls.jsonl", limit=pdf_limit)
+
+    wiki_files = get_all_files_paths_under(wikipedia_dir)
+    code_files = get_all_files_paths_under(github_dir)
+    pdf_files = get_all_files_paths_under(pdf_dir)
+
+    text_files = wiki_files + pdf_files
+
+    return text_files, code_files
+
+
+def run_curation_pipeline(args: Any, text_files: str, code_files: str) -> None:
+    """
+    Run the curation pipeline on the TinyStories dataset.
+
+    Args:
+        args (Any): Command-line arguments.
+        jsonl_dir (str): Directory path where the JSONL files are stored.
+    """
+    print("Running the curation pipeline...")
+    # Initialize the Dask cluster.
+    client = get_client(**parse_client_args(args))
+
+    # Define data curation steps for text and pdf files
+    curation_steps_text = Sequential(
+        [
+            dedupe,
+            filter_txt_lines,
+            filter_txt,
+            clean_and_unify,
+        ]
+    )
+
+    # Define data curation steps for code files
+    curation_steps_code = Sequential(
+        [
+            dedupe,
+            filter_code_lines,
+            filter_code,
+            clean_and_unify,
+            #          redact_pii,
```

Comment:
Remove?

Reply:
Lets keep this for now. I'm testing it even more for performance.

Reply:
Added it in.